### PR TITLE
Fix: Include documentation about needing to use buildscript for gradle plugins

### PR DESCRIPTION
**Pull request by AlexanderMann**
# Motivation

Spent an inordinate amount of time trying to get Cloud Spanner working alongside some custom socket stuff we needed to do for GCP's Cloud SQL IAM authentication. We wanted to use the same Gradle commands to make sure everything stayed in line with one another, and we were exercising the same code paths more frequently.

Unfortunately, Gradle is not my native tongue, and while I _knew_ that errors like the following _had_ to be related to the classpath, I couldn't for the life of me figure out how to resolve this, nor why the regular CLI worked out of the box just fine.

```
Caused by: org.flywaydb.core.api.FlywayException: Error occurred while executing flywayValidate
No database found to handle jdbc:cloudspanner:/projects/my-project/instances/dev/databases/database-name?credentials=/tmp/.appcreds.json
	at org.flywaydb.gradle.task.AbstractFlywayTask.runTask(AbstractFlywayTask.java:678)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
```

Eventually found [this comment in Github](https://github.com/flyway/flyway/issues/3355#issuecomment-1000585918) which solved everything. I'd like to think that had this existed in the documentation, I'd have gotten it on the first try 🤷 

## Notes

I am _sure_ my documentation here is prolly very off/I'm referring to things incorrectly in the Gradle sense. Happy to change things. Maybe @DoodleBobBuffPants has some thoughts on how to make this clearer (I simply copied their snippet afterall 😅 )

## Thank you!

Have used Flyway for years, love the tool. Thanks folks!

### CODE REVIEW

**Comment by DoodleBobBuffPants** [comment]
Thanks for the PR! We're currently working on V9 but will come back to this PR as soon as we can :)

File: documentation/usage/gradle/index.md [hunk 1]
```diff
@@ -176,6 +176,32 @@ flyway {
 
 For details on how to setup and use custom Gradle configurations, see the [official Gradle documentation](https://docs.gradle.org/current/dsl/org.gradle.api.artifacts.ConfigurationContainer.html).
 
+### Using Flyway Database Types
```
Comment (DoodleBobBuffPants) [inline documentation/usage/gradle/index.md]: Can you reword this to `Adding dependencies on Flyway Database Types` - it would be better to put `dependency` in some form in the header

File: documentation/usage/gradle/index.md [hunk 2]
```diff
@@ -176,6 +176,32 @@ flyway {
...
+
+For some Flyway database types (non core), like [Cloud Spanner](/documentation/database/cloud-spanner) or [SQL Server](/documentation/database/sqlserver), you'll need to actually use a `buildScript` to get your
+Gradle commands to work properly. You need to enable putting the database type on the build classpath, and not
+the project classpath.
+
+Something like the following should have you set:
```
Comment (DoodleBobBuffPants) [inline documentation/usage/gradle/index.md]: Can you reword this to ``Here is an example `build.gradle`:``

File: documentation/usage/gradle/index.md [hunk 3]
```diff
@@ -176,6 +176,32 @@ flyway {
...
+
+```groovy
+buildscript {
+    repositories {
+        mavenCentral()
+    }
+    dependencies {
+        classpath "org.flywaydb:flyway-mysql:${flywayVersion}"
```
Comment (DoodleBobBuffPants) [inline documentation/usage/gradle/index.md]: Can you replace `${flywayVersion}` with `{{ site.flywayVersion }}` which is the GitHub pages way of using our variable for the site version

File: documentation/usage/gradle/index.md [hunk 4]
```diff
@@ -176,6 +176,32 @@ flyway {
...
+    }
+}
+```
+
+Signs that you're running into this might be seeing errors like the following:
```
Comment (DoodleBobBuffPants) [inline documentation/usage/gradle/index.md]: Can you reword this to ``Without this you may see an error like the following: `No database found to handle jdbc...` `` and remove the code block below

File: documentation/usage/gradle/index.md [hunk 5]
```diff
@@ -176,6 +176,32 @@ flyway {
 
 For details on how to setup and use custom Gradle configurations, see the [official Gradle documentation](https://docs.gradle.org/current/dsl/org.gradle.api.artifacts.ConfigurationContainer.html).
 
+### Using Flyway Database Types
+
+For some Flyway database types (non core), like [Cloud Spanner](/documentation/database/cloud-spanner) or [SQL Server](/documentation/database/sqlserver), you'll need to actually use a `buildScript` to get your
+Gradle commands to work properly. You need to enable putting the database type on the build classpath, and not
+the project classpath.
```
Comment (DoodleBobBuffPants) [inline documentation/usage/gradle/index.md]: Could you reword this (keeping the links) to: ``For some Flyway database types, like Cloud Spanner and SQL Server you'll need to add a dependency to the database type in a `buildscript` closure to get your Gradle commands to work properly. This puts the database type on the build classpath, and not the project classpath.``

It's a bit more succinct and removes the `non core` component which isn't strictly true - we are looking at a long-term goal of having a module per database type in order to reduce the bloat in Flyway Core

**Review by DoodleBobBuffPants — CHANGES_REQUESTED** [review]

File: documentation/usage/gradle/index.md (hunk 5 above)
Comment (AlexanderMann) [reply documentation/usage/gradle/index.md]: Yeah, fwiw, I'd be _awesome_ to have the error message itself spit out _what_ database types _are_ installed and ready to go. Fighting with Gradle to see the classpath is :rage4: imo

(And happy to make the change 😌 )

**Comment by AlexanderMann** [comment]
@DoodleBobBuffPants 🏓 to you

**Review by DoodleBobBuffPants — APPROVED** [review]
Thanks for making these changes!

**Comment by DoodleBobBuffPants** [comment]
You've also been added to our [Hall of Fame](https://flywaydb.org/documentation/contribute/hallOfFame) :)

File: documentation/usage/gradle/index.md (hunk 5 above)
Comment (DoodleBobBuffPants) [reply documentation/usage/gradle/index.md]: Could you write that suggestion up as an issue [here](https://github.com/flyway/flyway/issues) in order to track it

**Comment by aelagon** [comment]
Hi all, 
should this kind of info be included also here -> https://flywaydb.org/documentation/usage/maven/ ?

I experienced the same issue when I tried to migrate using flyway 8 and JAVA api in a project built with Maven :)
